### PR TITLE
HUM-871 Fix Hummingbird to pass new Swift func tests

### PR DESCRIPTION
**Pull request by gholt**

### CODE REVIEW

File: proxyserver/middleware/versioned_writes.go [hunk 1]
```diff
@@ -334,7 +334,7 @@ func (v *versionedWrites) headObject(writer http.ResponseWriter, req *http.Reque
 
 func (v *versionedWrites) handleObjectDeleteStack(writer http.ResponseWriter, request *http.Request, account, container, versionsContainer, object string) {
 	ctx := GetProxyContext(request)
-	listingPath := fmt.Sprintf("/v1/%s/%s?format=json&prefix=%s&reverse=on", account, versionsContainer, v.versionedObjectPrefix(object))
+	listingPath := fmt.Sprintf("/v1/%s/%s?format=json&prefix=%s&reverse=on", common.Urlencode(account), common.Urlencode(versionsContainer), url.QueryEscape(v.versionedObjectPrefix(object)))
```
Comment (corystone) [inline proxyserver/middleware/versioned_writes.go]: Does this + containerListing() double encode?

Comment (corystone) [reply proxyserver/middleware/versioned_writes.go]: Oh that one doesn't? I can't read.

Comment (gholt) [reply proxyserver/middleware/versioned_writes.go]: :)

containerListing does not encode. That's why I changed its parameter name to urlStr. But yeah, all that diffy stuff gets confusing. :)

**Comment by corystone** [comment]
Thanks for fixing my bug! 👍

**Pull request merged**